src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpWindowsProxy.cs

```diff
@@ -166,6 +179,21 @@ private HttpWindowsProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessi
                     }
                 }
             }
+
+            _cachedProxies = new ConcurrentDictionary<string, CachedProxy>();
+            _cacheCleanupTimer = new Timer(state =>
+            {
+                var cachedProxies = (ConcurrentDictionary<string, CachedProxy>)state;
+                int ticks = Environment.TickCount;
+
+                foreach (KeyValuePair<string, CachedProxy> kvp in cachedProxies)
+                {
+                    if ((ticks - kvp.Value.LastAccessTicks) > CacheCleanupAgeMilliseconds)
+                    {
+                        cachedProxies.TryRemove(kvp.Key, out _);
```

Comment:
What happens if this thread checks LastAccessTicks and see that it's too old but then just before calling TryRemove, another thread accesses it?

Reply:
I didn't handle this race because the only side-effect will be, if a proxy fails, global failover will happen one request late.